### DISCUSSION TITLE
Pylance will not parse the code correctly when the try statement has more than 730 lines of code

### DISCUSSION THREAD

**Discussion by OnesoftQwQ** [microsoft/pylance-release#6789] [Q&A] [answered]
![图片](https://github.com/user-attachments/assets/90f22a1a-b4fb-4ff4-a803-c02dcb25f88d)

![图片](https://github.com/user-attachments/assets/72b58893-8a54-4643-8953-4cab329396e9)

**Comment by erictraut** [comment, marked as answer]
You have exceeded the cycolmatic complexity limit for a single function. Pyright (the static type checker that pylance is built upon) has internal limits to prevent performance issues when analyzing especially complex code flow graphs. Once this limit is exceeded, static analysis is skipped for that function.

You can avoid this by simplifying your code — by using loops or subroutines, for example.
